elements/castbar.lua

```diff
@@ -83,443 +84,292 @@ A default texture will be applied to the StatusBar and Texture widgets if they d
-	local shield = element.Shield
-	if(shield) then
-		shield:Hide()
+		element.duration = endTime - GetTime()
 	end
 
-	element.notInterruptible = nil
-
-	--[[ Callback: Castbar:PostCastInterruptible(unit)
-	Called after the element has been updated when a spell cast has become interruptible.
-
-	* self - the Castbar widget
-	* unit - unit for which the update has been triggered (string)
-	--]]
-	if(element.PostCastInterruptible) then
-		return element:PostCastInterruptible(unit)
+	if(delta < 0) then
+		delta = 0
 	end
-end
-
-local function UNIT_SPELLCAST_NOT_INTERRUPTIBLE(self, event, unit)
-	if(self.unit ~= unit and self.realUnit ~= unit) then return end
 
-	local element = self.Castbar
-	local shield = element.Shield
-	if(shield) then
-		shield:Show()
-	end
+	element.max = endTime - startTime
+	element.startTime = startTime
+	element.delay = element.delay + delta
 
-	element.notInterruptible = true
+	element:SetMinMaxValues(0, element.max)
+	element:SetValue(element.duration)
 
-	--[[ Callback: Castbar:PostCastNotInterruptible(unit)
-	Called after the element has been updated when a spell cast has become non-interruptible.
+	--[[ Callback: Castbar:PostCastUpdate(unit)
+	Called after the element has been updated when a spell cast has been updated.
 
 	* self - the Castbar widget
-	* unit - unit for which the update has been triggered (string)
+	* unit - the unit that the update has been triggered (string)
 	--]]
-	if(element.PostCastNotInterruptible) then
-		return element:PostCastNotInterruptible(unit)
+	if(element.PostCastUpdate) then
+		return element:PostCastUpdate(unit)
 	end
 end
 
-local function UNIT_SPELLCAST_DELAYED(self, event, unit)
-	if(self.unit ~= unit and self.realUnit ~= unit) then return end
+local function CastStop(self, event, unit, castID, spellID)
+	if(self.unit ~= unit) then return end
 
 	local element = self.Castbar
-	local name, _, _, startTime = UnitCastingInfo(unit)
-	if(not startTime or not element:IsShown()) then return end
-
-	local duration = GetTime() - (startTime / 1000)
-	if(duration < 0) then duration = 0 end
-
-	element.delay = element.delay + element.duration - duration
-	element.duration = duration
-
-	element:SetValue(duration)
-
-	--[[ Callback: Castbar:PostCastDelayed(unit, name)
-	Called after the element has been updated when a spell cast has been delayed.
-
-	* self - the Castbar widget
-	* unit - unit that the update has been triggered (string)
-	* name - name of the delayed spell (string)
-	--]]
-	if(element.PostCastDelayed) then
-		return element:PostCastDelayed(unit, name)
-	end
-end
 
-local function UNIT_SPELLCAST_STOP(self, event, unit, castID)
-	if(self.unit ~= unit and self.realUnit ~= unit) then return end
+	-- Channeled spells for some reason don't have castIDs
+	if(element.castID ~= castID or element.spellID ~= spellID) then return end
 
-	local element = self.Castbar
-	if(element.castID ~= castID) then
-		return
-	end
+	if(element.Spark) then element.Spark:Hide() end
```

Comment:
This might be needed in `CastFail` because of the `.timeToHold` option, but not here or do I miss something?

Reply:
There's no need to show the spark, when the bar is being set to its max value. It applies to both fails and stops, even the default castbars behave similarly.